src/Microsoft.ApplicationInsights/Managed/Shared/Extensibility/Implementation/Tracing/IHeartbeatProvider.cs

```diff
@@ -1,13 +1,20 @@
 ﻿namespace Microsoft.ApplicationInsights.Extensibility.Implementation.Tracing
 {
     using System;
-    using System.Collections.Generic;
     using Microsoft.ApplicationInsights.Extensibility;
 
+    /// <summary>
+    /// Internal interface for providing heartbeat information into the SDK pipeline. Useful
+    /// for mocking for unit test purposes.
+    /// </summary>
     internal interface IHeartbeatProvider : IHeartbeatPropertyManager, IDisposable
     {
         string InstrumentationKey { get; set; }
 
         void Initialize(TelemetryConfiguration configuration);
+
+        bool AddHeartbeatProperty(string propertyName, bool overrideDefaultField, string propertyValue, bool isHealthy);
+
+        bool SetHeartbeatProperty(string propertyName, bool overrideDefaultField, string propertyValue = null, bool? isHealthy = null);
```

Comment:
mentioned somewhere else too - if this is the only one used, why we have 3 arg version of this same method

Reply:
Internal/external access. Please see the above discussion.